common/src/main/java/com/wynntils/models/containers/BankModel.java

```diff
@@ -68,15 +68,15 @@ public void onScreenInit(ScreenInitEvent e) {
         if (!(e.getScreen() instanceof AbstractContainerScreen<?> screen)) return;
 
         if (Models.Container.isAccountBankScreen(screen)) {
-            currentContainer = SearchableContainerType.ACCOUNT_BANK;
+            currentContainer = InteractiveContainerType.ACCOUNT_BANK;
         } else if (Models.Container.isBlockBankScreen(screen)) {
-            currentContainer = SearchableContainerType.BLOCK_BANK;
+            currentContainer = InteractiveContainerType.BLOCK_BANK;
         } else if (Models.Container.isBookshelfScreen(screen)) {
-            currentContainer = SearchableContainerType.BOOKSHELF;
+            currentContainer = InteractiveContainerType.BOOKSHELF;
```

Comment:
I see that the enum already has a pattern for the screens. Is that necessary, seeing that most of them are seemingly also in `ContainerModel`? 

I think we have two options here:
1, Move all patterns to `ContainerModel`, add `Predicate<Screen>` for the enum
2, Remove all duplicated patterns from `ContainerModel`, and only keep them in `InteractiveContainerType`. Adding a `Predicate<Screen>` for enums is trivial in this case.

I would lean towards the second option, as I've never liked `ContainerModel` having so much patterns + is...Screen methods. I think we could create two enums to fit all purposes. `InteractiveScreenType` and `StaticScreenType` (or what fits best here) and store all related info that way.

However, let's consult with @magicus before doing this change (and feel free to do the other suggestions I've recommended).

Reply:
I would need to study the code to know exactly what I'm talking about (and I'm running out of time rn, sorry), but I think I also prefer 2 here, if I understand it correctly. I never liked all the patterns, it felt very much like an anti-pattern (no pun intended). Somehow we need to bundle everything we know about a container into a separate class, patterns to detect it, what buttons to press to scroll it, etc. I think that is what 2 says.

Reply:
Quickly scrolling through the code... let me add another note: I don't think we should have an *enum* at all. We should have subclasses and/or instances that represent everything we need to know about a screen. (Regardless of if we can interact with it or not, just as long as wee need to know what kind of screen it is.) 

If this PR is the place to make such a fix, I don't know.

Reply:
I've done the predicate for now, I've also moved a lot of logic out of ContainerModel, especially some that was more relevant to BankModel.

I think the idea of having subclasses/instances to contain all relevant information about any screen we do anything with is interesting, probably not something I'd do here but definitely worth looking into it at some point

Reply:
I think it's fine to have the enum in this PR, since it's already a major code improvement. We can separate it to instances later, when we have a solid use case for it.